### PR TITLE
b2c-configurable-bundle-templates

### PR DESCRIPTION
**Pull request by gorav-sharma-metacube**
CC-16231: b2c-configurable-bundle-templates

### CODE REVIEW

File: tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    And Each array element of array in response should contain property with value:    [data][relationships][configurable-bundle-template-slots][data]    type    configurable-bundle-template-slots
+    And Each array element of array in response should contain property:    [data][relationships][configurable-bundle-template-slots][data]    id
+    And Response should contain the array of a certain size:    [included]    706
+    And Response body parameter should be:    [included][0][type]    concrete-product-image-sets
+    And Response body parameter should not be EMPTY:    [included][0][id]
+    And Response body parameter should be:    [included][0][attributes][imageSets][0][name]    default
+    And Each array element of array in response should contain property:    [included][0][attributes][imageSets][0][images]    externalUrlLarge
+    And Each array element of array in response should contain property:    [included][0][attributes][imageSets][0][images]   externalUrlSmall
+    And Response include element has self link:   concrete-product-image-sets
+    And Response body parameter should be:    [included][1][type]    concrete-product-prices
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot:115]: please do not hardcore indexes. use standard way to check includes like i the test above

Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot]: fixed, i checked all the common data in indexes.

File: tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    And Each array element of array in response should contain property with value:    [data][relationships][configurable-bundle-template-image-sets][data]    type    configurable-bundle-template-image-sets
+    And Each array element of array in response should contain property:    [data][relationships][configurable-bundle-template-slots][data]    id
+    And Response body has correct self link internal
+    And Each array element of array in response should contain property:    [included]    id
+    And Each array element of array in response should contain nested property:    [included]    attributes    name
+    And Response body parameter should be:    [included][0][type]    configurable-bundle-template-slots
+    And Response body parameter should be:    [included][1][type]    configurable-bundle-template-slots
+    And Response body parameter should be:    [included][2][type]    configurable-bundle-template-slots
+    And Response body parameter should be:    [included][3][type]    configurable-bundle-template-slots
+    And Response body parameter should be:    [included][4][type]    configurable-bundle-template-image-sets
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot:92]: why is this written like this? change to:
And Response include should contain certain entity type:    configurable-bundle-template-slots
And Response include should contain certain entity type:    configurable-bundle-template-image-sets

File: tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response body parameter should be:    [data][type]    configurable-bundle-templates
+    And Response body parameter should be:    [data][id]    ${configurable_bundle_template_id}
+    And Response body parameter should be:    [data][attributes][name]   ${configurable_bundle_template_name_1}
+    And Response body parameter should not be EMPTY:    [data][links][self]
+    And Response should contain the array larger than a certain size:    [data][relationships][configurable-bundle-template-slots][data]    3
+    And Each array element of array in response should contain property with value:    [data][relationships][configurable-bundle-template-slots][data]    type    configurable-bundle-template-slots
+    And Each array element of array in response should contain property:    [data][relationships][configurable-bundle-template-slots][data]    id
+    And Response should contain the array of a certain size:    [included]    706
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot:108]: we do not need this check

File: tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    And Each array element of array in response should contain property with value in:    [included]    type    configurable-bundle-template-slots    concrete-product-image-sets    concrete-product-prices    concrete-products
+    And Each array element of array in response should contain property with value NOT in:    [included]    id    None
+    And Each array element of array in response should contain property with value NOT in:    [included]    name    None
+    And Each array element of array in response should contain nested property:    [included]    [attributes]    name
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot:112]: the above 4 lines should be replaced with 

And Response include should contain certain entity type: configurable-bundle-template-slots
And Response include should contain certain entity type: concrete-product-image-sets 
And Response include should contain certain entity type: concrete-product-prices
And Response include should contain certain entity type: concrete-products

Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot]: fixed

File: tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot (hunk 3 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot]: fixed

File: tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot (hunk 2 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/configured_bundles_endpoints/configured_bundles_templates/positive.robot]: fixed

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**